ourjapanlife/findadoc-web
# feat: add validation to submission form

Part of #502 

## 🔧 What changed
This PR adds validation to the fields that appear on the Submit page. This will help to prevent users from submitting incomplete or invalid data. The changes include:

1. Add validation checks to the fields.
2. Show messages above fields that don't pass validation. Messages are localized in English and Japanese.
3. Prevent the Submit button from working if validation checks don't pass.
4. Change the optional Notes input to a resizable text area to improve readability for longer messages.

## 🧪 Testing instructions
1. Go to `/submit` after running the frontend server
2. Confirm that the validation text appears when fields have invalid data, and disappears when the fields have valid data
3. Click the Submit button when the form has invalid data, and confirm that the form doesn't submit.
5. Click the Submit button when the form has valid data, and confirm that the Success screen is shown.

## 📸 Screenshots

-   ### Before
![Screenshot 2024-06-04 at 11 04 21 PM](https://github.com/ourjapanlife/findadoc-web/assets/31802656/c430dc31-7989-4efd-8d4a-c6c3e9494a57)


-   ### After
![Screenshot 2024-06-04 at 11 04 03 PM](https://github.com/ourjapanlife/findadoc-web/assets/31802656/0f86a90b-d269-45da-92ad-ccae25728870)
